boxoffice/forms/discount.py

```diff
@@ -0,0 +1,65 @@
+# -*- coding: utf-8 -*-
+
+from baseframe import __
+import baseframe.forms as forms
+from baseframe.forms.sqlalchemy import QuerySelectMultipleField, QuerySelectField
+from coaster.utils import getbool
+from ..models import DISCOUNT_TYPE, DiscountPolicy, CURRENCY
+
+__all__ = ['DiscountPolicyForm', 'DiscountPriceForm', 'DiscountCouponForm']
+
+
+class DiscountPolicyForm(forms.Form):
+    title = forms.StringField(__("Discount title"),
+        validators=[forms.validators.DataRequired(__("Please specify a discount title")),
+        forms.validators.Length(max=250), forms.validators.StripWhitespace()])
+    is_price_based = forms.RadioField(__("Price based discount"),
+        choices=[(True, __("Special price discount")),
+        (False, __("Percentage based discount"))], coerce=getbool, default=True)
+    discount_type = forms.RadioField(__("Discount Type"),
+        choices=DISCOUNT_TYPE.items(), coerce=int, default=DISCOUNT_TYPE.values()[1])
+    percentage = forms.IntegerField(__("Percentage"),
+        validators=[forms.validators.OptionalIf('is_price_based', __("Please specify a discount percentage"))])
+    item_quantity_min = forms.IntegerField(__("Minimum number of tickets"), default=1)
+    discount_code_base = forms.NullTextField(__("Discount Title"),
+        validators=[forms.validators.OptionalIfNot('discount_type', __("Please specify a discount code base")),
+        forms.validators.Length(max=20), forms.validators.StripWhitespace()])
+    bulk_coupon_usage_limit = forms.IntegerField(__("Number of times a bulk coupon can be used"), default=1)
+    items = QuerySelectMultipleField(__("Items"), get_label='title', query_factory=lambda: [],
```

Comment:
Why should this be mandatory only if the discount policy is price based? I think it makes sense to make `items` mandatory regardless.